src/drawer.canvasentry.js

```diff
@@ -145,40 +145,65 @@ export default class CanvasEntry {
 
     /**
      * Set the fill styles for wave and progress
-     * @param {string|string[]} waveColor Fill color for the wave canvas,
-     * or an array of colors to apply as a gradient
-     * @param {?string|string[]} progressColor Fill color for the progress canvas,
-     * or an array of colors to apply as a gradient
+     * @param {string|string[]|CanvasImageSource} waveColor Fill color for
+     * the wave canvas, an array of colors to apply as a gradient, or a Canvas
+     * Image Source (HTML Image, SVG Image, HTML Video, HTML Canvas, Image
+     * Bitmap and Offscreen Canvas elements)
+     * @param {?string|string[]|CanvasImageSource} progressColor Fill color for
+     * the wave canvas, an array of colors to apply as a gradient, or a Canvas
+     * Image Source (HTML Image, SVG Image, HTML Video, HTML Canvas, Image
+     * Bitmap and Offscreen Canvas elements)
+     * @param {object} waveStyleOptions Wave style options
+     * @param {object} progressStyleOptions Progress style options
      */
-    setFillStyles(waveColor, progressColor) {
-        this.waveCtx.fillStyle = this.getFillStyle(this.waveCtx, waveColor);
+    setFillStyles(waveColor, progressColor, waveStyleOptions, progressStyleOptions) {
```

Comment:
what about using `setFillStyles(waveStyleOptions, progressStyleOptions)` and extract the colors from the options. `setWaveColor` etc would have to be adjusted as well.

Reply:
so the waveColor parameter would accept an object containing: the "style" (color, gradient, etc..) as a property, and the options for that style as a property?